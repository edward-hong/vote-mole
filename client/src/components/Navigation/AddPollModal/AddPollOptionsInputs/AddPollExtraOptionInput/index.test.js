import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import AddPollExtraOptionInput from '.'

describe('AddPollExtraOptionInput', () => {
	it('shows error message when there is an error', () => {
		const error = 'Please fill out all options fields'
		const { getByText } = render(
			<AddPollExtraOptionInput
				input={{ name: 'field' }}
				meta={{ touched: true, error }}
				index={0}
				deleteOption={() => {}}
			/>,
		)

		expect(getByText(error)).toHaveTextContent(error)
	})

	it('renders correctly when valid', () => {
		const { container } = render(
			<AddPollExtraOptionInput
				input={{ name: 'field' }}
				meta={{ touched: true, error: false }}
				index={0}
				deleteOption={() => {}}
			/>,
		)

		expect(container.firstChild).toMatchSnapshot()
	})

	it('renders correctly when untouched', () => {
		const { container } = render(
			<AddPollExtraOptionInput
				input={{ name: 'field' }}
				meta={{ touched: false, error: false }}
				index={0}
				deleteOption={() => {}}
			/>,
		)

		expect(container.firstChild).toMatchSnapshot()
	})

	it('calls delete handler when delete button is clicked', () => {
		const deleteOption = jest.fn()
		const index = 0
		const { container } = render(
			<AddPollExtraOptionInput
				input={{ name: 'field' }}
				meta={{ touched: false, error: false }}
				index={index}
				deleteOption={deleteOption}
			/>,
		)
		const deleteButton = container.getElementsByTagName('button')[0]

		fireEvent.click(deleteButton)

		expect(deleteOption).toHaveBeenCalledWith(index)
	})
})
