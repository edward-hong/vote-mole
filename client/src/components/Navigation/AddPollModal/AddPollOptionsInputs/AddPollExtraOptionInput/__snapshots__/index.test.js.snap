// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPollExtraOptionInput renders correctly when untouched 1`] = `
.c0 {
  margin-bottom: 3px;
}

.c0 input {
  height: 47px;
}

<div
  class="c0 input-group"
>
  <input
    class="form-control"
    id="field"
    name="field"
    type="text"
  />
  <div
    class="input-group-append"
  >
    <button
      class="btn btn-outline-danger"
      type="button"
    >
      <span
        aria-hidden="true"
        class="fa fa-close fa-2x"
      />
    </button>
  </div>
</div>
`;

exports[`AddPollExtraOptionInput renders correctly when valid 1`] = `
.c0 {
  margin-bottom: 3px;
}

.c0 input {
  height: 47px;
}

<div
  class="c0 input-group"
>
  <input
    class="is-valid form-control"
    id="field"
    name="field"
    type="text"
  />
  <div
    class="input-group-append"
  >
    <button
      class="btn btn-outline-danger"
      type="button"
    >
      <span
        aria-hidden="true"
        class="fa fa-close fa-2x"
      />
    </button>
  </div>
</div>
`;
